### PR TITLE
Travis Insights - Spotlight

### PR DESCRIPTION
**Pull request by chinmaychaudhari-accolite**

### CODE REVIEW

**Review by vitalied — CHANGES_REQUESTED** [review]
@chinmaychaudhari-accolite pls reformat code where needed:
- 2 spaces ident instead of 4.
- empty new line of the end of files

File: app/styles/app/layouts/travis-ci-insights/aggregate-matrices.scss [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+}
+
+.border-color {
+  border-color: #cce1ff;
+}
+
+.heading {
+  font-weight: bolder;
+  font-size: x-large;
+}
```
Comment (chinmaychaudhari-accolite) [inline app/styles/app/layouts/travis-ci-insights/aggregate-matrices.scss:27]: new line added

File: app/index.html [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
+      src="//cdn.jsdelivr.net/cal-heatmap/3.5.2/cal-heatmap.min.js"
+    ></script>
 
     {{content-for "head-footer"}}
   </head>
   <body>
     <noscript>
-      <div style="width: 60%; margin: auto;">
-        <img src="/images/logos/TravisCI-Mascot-1.png" alt="Travis CI mascot" width="200" style="float: left; margin: 1em 3em;">
+      <div style="width: 60%; margin: auto">
```
Comment (vitalied) [inline app/index.html:46]: revert

File: app/index.html [hunk 3]
```diff
@@ -1,34 +1,62 @@
...
+          alt="Travis CI mascot"
+          width="200"
+          style="float: left; margin: 1em 3em"
+        />
         <div>
           <h1>Hey there!</h1>
           <p>Looks like you have JavaScript disabled.</p>
           <p>The Travis CI webclient needs JavaScript to work properly.</p>
-          <h2>Please enable JavaScript to get the best Travis CI experience.</h2>
+          <h2>
```
Comment (vitalied) [inline app/index.html:57]: revert

File: app/index.html [hunk 4]
```diff
@@ -39,6 +67,5 @@ <h3>Thank you!</h3>
     <script src="{{rootURL}}assets/travis.js"></script>
 
     {{content-for "body-footer"}}
-
```
Comment (vitalied) [inline app/index.html:42]: revert

File: app/router.js [hunk 5]
```diff
@@ -8,6 +8,7 @@ const Router = EmberRouter.extend({
 
 Router.map(function () {
   this.route('confirm-user', { path: '/confirm-user/:token' });
+  this.route('travis-ci-insights');
```
Comment (vitalied) [inline app/router.js:11]: remove. There are already routes t the end of this file

File: app/styles/app.scss [hunk 6]
```diff
@@ -97,6 +98,7 @@
 @import "app/layouts/insights";
 @import "app/layouts/insights/insights";
 @import "app/layouts/billing";
+@import "app/layouts/travis-ci-insights/travis-ci-insights";
```
Comment (vitalied) [inline app/styles/app.scss:101]: reuse the ones from lines 98 or 99

File: app/styles/app.scss [hunk 7]
```diff
@@ -22,6 +22,7 @@
 @import "app/base";
 @import "app/z-index";
 @import "app/typography";
+@import "app/heatmap";
```
Comment (vitalied) [inline app/styles/app.scss:25]: rename to `insights-heatmap` and put after line 99

File: app/index.html [hunk 8]
```diff
@@ -1,34 +1,62 @@
 <!DOCTYPE html>
 <html>
   <head>
-    <meta charset="utf-8">
-    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta charset="utf-8" />
```
Comment (vitalied) [inline app/index.html:4]: revert

File: app/index.html [hunk 9]
```diff
@@ -1,34 +1,62 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
     <title>Travis CI</title>
-    <meta name="description" content="Travis CI enables your team to test and ship your apps with confidence. Easily sync your projects with Travis CI and you'll be testing your code in minutes.">
-    <meta name="viewport" content="width=device-width, initial-scale=1">
+    <meta
```
Comment (vitalied) [inline app/index.html:7]: revert

File: app/index.html [hunk 10]
```diff
@@ -1,34 +1,62 @@
...
     <script src="/check-browser-support.js" async></script>
 
     {{content-for "head"}}
 
-    <link href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:300,400,600,700" rel="stylesheet" type="text/css">
-    <link href="https://fonts.googleapis.com/css?family=Cousine:400,400i,700,700i" rel="stylesheet" type="text/css">
-    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/vendor.css">
-    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/travis.css">
-    <link rel="mask-icon" href="images/favicon.svg" color="black">
+    <link
```
Comment (vitalied) [inline app/index.html:18]: revert

File: app/index.html [hunk 11]
```diff
@@ -1,34 +1,62 @@
...
+      href="https://fonts.googleapis.com/css?family=Cousine:400,400i,700,700i"
+      rel="stylesheet"
+      type="text/css"
+    />
+    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/vendor.css" />
+    <link integrity="" rel="stylesheet" href="{{rootURL}}assets/travis.css" />
+    <link rel="mask-icon" href="images/favicon.svg" color="black" />
+
+    <link
```
Comment (vitalied) [inline app/index.html:32]: put css and js into assets

**Review by vitalied — CHANGES_REQUESTED** [review]

File: app/styles/app.scss (hunk 7 above)
Comment (prembalaji39) [reply app/styles/app.scss]: renamed to insights-heatmap and added after line 99

File: app/router.js (hunk 5 above)
Comment (RaviRanjan86) [reply app/router.js:11]: At the end of this there routes for insights which was already present in the past also. we do have added new route travis-ci-insights and there is only one instance of this in this whole file

File: app/index.html (hunk 2 above)
Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

File: app/index.html (hunk 3 above)
Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

File: app/index.html (hunk 8 above)
Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

File: app/index.html (hunk 9 above)
Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

File: app/index.html (hunk 10 above)
Comment (prembalaji39) [reply app/index.html]: formatted the code and indentation.

**Comment by vitalied** [comment]
@RaviRanjan86 pls fix conflicts

**Comment by RaviRanjan86** [comment]
Conflict resolved

**Comment by vitalied** [comment]
@shairyar can you do also a quick review?

File: app/router.js (hunk 5 above)
Comment (vitalied) [reply app/router.js:11]: @RaviRanjan86 why you marked this as resolved?
The page should be rendered under /insights URL, and its tab should be before "Notifications" tab. You should re-use this route:
`this.route('insights', { resetNamespace: true }, function () {`

File: .jsintrc [hunk 12]
```diff
@@ -0,0 +1,3 @@
+{
+    "predef":["daterangepicker"]
+}
```
Comment (vitalied) [inline .jsintrc:3]: @RaviRanjan86 new line at the end

**Review by vitalied — CHANGES_REQUESTED** [review]

**Review by travis-architect — APPROVED** [review]

**Comment by travis-architect** [comment]
@AndriiMysko @RaviRanjan86 @ the build is failing, can you please have a look?

**Comment by AndriiMysko** [comment]
> @AndriiMysko @RaviRanjan86 @ the build is failing, can you please have a look?

Hi, I am not part of this project. The only commit that I co-authored in this PR (first one) has passing build.

**Comment by RaviRanjan86** [comment]
> @AndriiMysko @RaviRanjan86 @ the build is failing, can you please have a look?

Hi @travis-architect, we have checked the build. All the errors in the build are not related to any of the files that we have added or modified. The test cases failing are not from the test cases that we have added recently. Please refer below logs url.

https://app.travis-ci.com/github/travis-ci/travis-web/builds/247845768

File: app/styles/app/layouts/spotlight/insights-heatmap.scss [hunk 13]
```diff
@@ -0,0 +1,103 @@
...
+.all {
+  background-color: #04c2bf;
+}
+.successful {
+  background-color: #32ab52;
+}
+.errored {
+  background-color: #db5c07;
+}
+.canceled {
```
Comment (vitalied) [inline app/styles/app/layouts/spotlight/insights-heatmap.scss:75]: @chinmaychaudhari-accolite why you are using generic names for classes?
the UI is broken!!!

**Review by vitalied — CHANGES_REQUESTED** [review]

File: app/styles/app/layouts/spotlight/insights-heatmap.scss (hunk 13 above)
Comment (prembalaji39) [reply app/styles/app/layouts/spotlight/insights-heatmap.scss]: Heatmap css generic names are removed and changed.

Comment (prembalaji39) [reply app/styles/app/layouts/spotlight/insights-heatmap.scss]: Screenshot from 2022-03-15 12-37-33

**Comment by vitalied** [comment]
@prembalaji39 pls rebase and fix conflicts

**Pull request closed without merging**